42 crear tests para el servicio de openlibrary

paul-ot opened the pull request:
Trello card: https://trello.com/c/nfv8yf0Q/42-crear-tests-para-el-servicio-de-openlibrary

Details:
- Added Webmock gem.
- Stubs for success and not found.
- Added tests for success and not found books.

paul-ot committed:
partial changes

paul-ot committed:
more partial changes

paul-ot committed:
added 2 tests, for success and not found, the stubs are in the same file

holywyvern commented on spec/services/open_library_spec.rb:
```diff
@@ -0,0 +1,35 @@
+
+  it 'gets a book not found error from open library' do
+    uri = URI('https://openlibrary.org/api/books?bibkeys=ISBN:0385472579abc&format=json&jscmd=data')
+    response = Net::HTTP.get(uri)
```
No funciona si haces directamente `OpenLibrary.new.find_by_isbn(isbn)` ?
Porque estás probando Net::HTTP en lugar de tu servicio de esta forma.

O sea, se que usas HTTPParty pero de fondo, HTTPParty usa Net::HTTP de base.

paul-ot replied:
Tenés razón, ahí lo cambié.

paul-ot committed:
- Changed the Net::HTTP.get for OpenLibrary.new.find_by_isbn.

holywyvern commented on spec/services/open_library_spec.rb:
```diff
@@ -0,0 +1,33 @@
+  end
+
+  it 'gets a book not found error from open library' do
+    response = OpenLibrary.new.find_by_isbn('0385472579abc')
```
Deberías hacer is_expected to_throw(OpenLibraryException), en lugar de atrapar el error y no hacer nada

holywyvern replied:
o `expect { OpenLibrary.new.find_by_isbn('0385472579')  }.to throw(OpenLibraryException)`

paul-ot replied:
Fixed.

paul-ot committed:
Implementing the use of:

expect { #... }.to raise_error(OpenLibraryException)

Instead of catching an error.

paul-ot committed:
Merge branch 'development' into 42-crear-tests-para-el-servicio-de-openlibrary

paul-ot commented:
Done, @holywyvern .
